actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchNotificationActor.java

```diff
@@ -0,0 +1,361 @@
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.actors.notificationservice.EmailServiceActor;
+import org.sunbird.learner.util.Util;
+import org.sunbird.models.course.batch.CourseBatch;
+
+/**
+ * This actor will initiates email notification when user get enrolled/unenrolled to a open batch
+ * and also for Mentor/Participants addition/removal for an "invite-only" batches
+ *
+ * @author github.com/iostream04
+ */
+@ActorConfig(
+  tasks = {"courseBatchNotification"},
+  asyncTasks = {"courseBatchNotification"}
+)
+public class CourseBatchNotificationActor extends BaseActor {
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+
+  private static EmailServiceClient emailServiceClient = EmailServiceFactory.getInstance();
+  private static final Props props = Props.create(EmailServiceActor.class);
+  private static ActorSystem system = ActorSystem.create("system");
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+
+    String requestedOperation = request.getOperation();
+    if (requestedOperation.equals(ActorOperations.COURSE_BATCH_NOTIFICATION.getValue())) {
+      courseBatchNotication(request);
+    } else {
+      ProjectLogger.log(
+          "CourseBatchNotificationActor : onReceive, No suitable Operation found for the request "
+              + request.getOperation(),
+          LoggerEnum.ERROR);
+    }
+  }
+
+  private void courseBatchNotication(Request request) {
+    Map<String, Object> requestMap = request.getRequest();
+    if (requestMap.get(JsonKey.USER_ID) != null) {
+      batchEnrollOperationNotifier(
+          (String) requestMap.get(JsonKey.USER_ID),
+          (CourseBatch) requestMap.get(JsonKey.COURSE_BATCH),
+          (String) requestMap.get(JsonKey.OPERATION_TYPE));
+    } else if (requestMap.get(JsonKey.UPDATE) != null) {
+      batchUpdateOperationNotifier(
+          (CourseBatch) requestMap.get(JsonKey.COURSE_BATCH),
+          (Map<String, Object>) requestMap.get(JsonKey.USERIDS));
+    } else if (requestMap.get(JsonKey.OPERATION_TYPE) != null) {
+      batchCreateOperationNotifier(
+          (CourseBatch) requestMap.get(JsonKey.COURSE_BATCH),
+          (String) requestMap.get(JsonKey.OPERATION_TYPE));
+    }
+  }
+
+  /*
+   * This method process invite-only batch enrol and unenroll operation
+   * notifications mentor and learner add remove Operation
+   *
+   * @param CourseBatch object for getting course and email related data
+   *
+   * @param operationType String for specifying the operation type ie: add/remove
+   */
+  private void batchCreateOperationNotifier(CourseBatch courseBatch, String operationType) {
+    if (operationType.equals(JsonKey.ADD)) {
```

Comment:
Unnecessary if check. Directly pass operationType. This method not event required then.